Added popup handling for popups and layer shell surfaces

Gaspard-- opened the pull request:
Need to test this before merging.

Gaspard-- committed:
Added popup handling for popups and layer shell surfaces

Gaspard-- commented:
Going to go ahead and add focus handling: it should simplify the testing of popups.

riamon-v reviewed (commented):
Seems not to work

Gaspard-- committed:
Refactored surface workspace handling

riamon-v commented:
> 
> 
> Going to go ahead and add focus handling: it should simplify the testing of popups.

riamon-v reviewed (commented):
I tried to open Libre office, which has a popup window at the start and this window seems to be part of the Window Tree. However when we're trying to go in the settings of Firefofx for example these popups are actually working. The implementation should've been missunderstood maybe

Gaspard-- committed:
Added focus handling basics

Gaspard-- committed:
Improved popoup input and focus handling.

A couple of choices were made, openned issues corresponding to the possible albinos configuration.

Gaspard-- commented:
Couple of comments to help review:
- to simplify things a bit, added a `getFocusedView` in server.
- getFocusedSurface is similar to getFocusedView, but also takes other surfaces into account, not only desktop views.
- View::view_at likewise checks layer & xdg views
- XdgView::desktop_view_at is not really used yet, but only checks desktop views. (I think we'll probably need it when cleaning multiple output handling)
- You will often see the pattern where I check a range of layers, then the desktop and then sometimes another range of layers. This is because ZWLR_LAYER_SHELL_V1_LAYER_OVERLAY and ZWLR_LAYER_SHELL_V1_LAYER_TOP are above desktop surfaces, and the 2 others below.
- Keyboard interactive layer surfaces below the desktop view aren't supported. The others should steal all focus (which is why View::view_at is shortcircuited if one is active)

riamon-v commented on include/Popup.hpp:
```diff
@@ -14,15 +14,14 @@ struct PopupListeners
-    void handle_new_popup(wl_listener *listener, void *data);
-    void handle_destroy_popup(wl_listener *listener, void *data);
+  wlr_surface *surface;
+  std::unique_ptr<Popup> popup;
```
Is this in the case where a popup can have its own popups ?

Gaspard-- replied:
Yes. I deduced that should be the case since handle_new_popup exists.

riamon-v commented on include/Server.hpp:
```diff
@@ -54,6 +54,15 @@ class Server
+    return outputManager.getActiveWorkspace()->getViews().front().get();
+  }
+    
+  wlr_surface *getFocusedSurface() const noexcept;
```
Why is this function not implemented like the others in this file ?

Gaspard-- replied:
I thought it was likelly to change implementation, so I wanted to avoid having systematic compilation.
